src/dune_rules/coq/coq_rules.ml

```diff
@@ -408,12 +406,19 @@ let setup_coqc_rule ~loc ~dir ~sctx ~coqc_dir ~file_targets ~stanza_flags
     >>| Action.Full.add_sandbox Sandbox_config.no_sandboxing)
 
 let coq_modules_of_theory ~sctx lib =
+  (* TODO generalize to any coq library (even installed) *)
   Action_builder.of_memo
-    (let name = Coq_lib.name lib in
-     let dir = Coq_lib.src_root lib in
-     let* dir_contents = Dir_contents.get sctx ~dir in
-     let+ coq_sources = Dir_contents.coq dir_contents in
-     Coq_sources.library coq_sources ~name)
+  @@
+  let name = Coq_lib.name lib in
+  Path.as_in_build_dir (Coq_lib.src_root lib) |> function
+  | None ->
+    (* TODO for now we return no modules when the path of a theory is outside of
+       the build directory *)
+    Memo.return []
```

Comment:
We need to fix this, it is a bit silly as it is only used by coqdep, but we need to keep track of the digest of the user-contrib contents as to rerun coqdep if that changes, due to possible loadpath resolution differences.

It seems easy to solve fortunately.

Reply:
Not sure how to solve this yet.

Reply:
OK I've added some detection for coq modules in an install location.